### PR TITLE
Remove file from localstorage

### LINKED ISSUE #73: Remove `file://`
To implement the benchmarks, we need to remove the `file://` scheme (which also isn't allowed by the htsget spec). See [here](https://github.com/umccr/htsget-rs/blob/ce002ef5ed4d97b2d721b89e73b34baa0d364625/htsget-search/src/storage/blocking/local.rs#L80). Last time I just changed it for `http://localhost`, but this time I was thinking maybe we should take a more permanent route? If I understand correctly, `LocalStorage` shouldn't be constrained to localhost.

### PR DESCRIPTION
**Pull request by mmalenic**
Closes #73 

The following PR removes the association of `file://` from htsget response tickets, as this is not allowed by the protocol, and replaces them with `https://`.

Changes:
* Add a new trait called `UrlFormatter` for formatting urls, implementing it with `HttpsFormatter` for formatting urls using the https scheme.
* Add a static file server called `AxumStorageServer`, that can actually respond to the urls generated by `HttpsFormatter`, based on the axum library.
* `AxumStorageServer` supports TLS using certificates that can be specified with htsget-config.
* `HttpsFormatter` used by `LocalStorage` to format urls.
* `AxumStorageServer` spawned in htsget-actix-web alongside main server which responds to queries.
* Add test cases for the server and formatter.
* Change all relevant test cases expecting `file://` to expecting `https://`.
* A few more minor code changes and clean up.

### CODE REVIEW

File: htsget-config/src/config.rs [hunk 1]
```diff
@@ -67,6 +75,10 @@ pub struct Config {
   pub htsget_updated_at: Option<String>,
   pub htsget_environment: Option<String>,
   pub htsget_s3_bucket: Option<String>,
+  pub htsget_localstorage_ip: String,
+  pub htsget_localstorage_port: String,
```
Comment (brainstorm) [inline htsget-config/src/config.rs:79]: @mmalenic Reviewing this, not sure it's ready to merge but I just noticed that ports/ips still don't have the more fitting networking types?

File: htsget-http-core/src/lib.rs [hunk 2]
```diff
@@ -132,7 +133,7 @@ mod tests {
       Ok(JsonResponse::from_response(Response::new(
         Format::Bam,
         vec![Url::new(format!(
-          "file://{}",
+          "https://127.0.0.1:8081{}",
```
Comment (brainstorm) [inline htsget-http-core/src/lib.rs:136]: I guess those hardcoded ip/port is the default just for testing?

**Review by brainstorm — APPROVED** [review]
Thanks for getting rid of the local file:// scheme, it was a bit more involved than expected but it's looking good!

File: htsget-http-core/src/lib.rs (hunk 2 above)
Comment (mmalenic) [reply htsget-http-core/src/lib.rs:136]: Yeah I just added those for testing. Potentially we could introduce some constants at the top of test modules which refactor hardcoded values, as there is a bit of repetition. But I'm not sure if it's is worth it?

File: htsget-config/src/config.rs (hunk 1 above)
Comment (mmalenic) [reply htsget-config/src/config.rs:79]: I was planning on changing those in the next PR, where I edit some of the config stuff. For example, a user should be able to choose storage type (LocalStorage vs S3 Storage), which is not yet possible. That should be a fairly small PR.

File: htsget-http-core/src/lib.rs (hunk 2 above)
Comment (brainstorm) [reply htsget-http-core/src/lib.rs:136]: If those are just for the testing bits, nevermind 👍🏻

File: htsget-config/src/config.rs (hunk 1 above)
Comment (brainstorm) [reply htsget-config/src/config.rs:79]: Gotcha, looks like this PR doesn't break anything, so I would keep merging otherwise things pile up! ;)

**Pull request merged**